detectors/node/opentelemetry-resource-detector-alibaba-cloud/src/detectors/AlibabaCloudEcsDetector.ts

```diff
@@ -57,26 +58,40 @@ class AlibabaCloudEcsDetector implements Detector {
    *
    * @param config (unused) The resource detection config
    */
-  async detect(_config?: ResourceDetectionConfig): Promise<Resource> {
-    const {
-      'owner-account-id': accountId,
-      'instance-id': instanceId,
-      'instance-type': instanceType,
-      'region-id': region,
-      'zone-id': availabilityZone,
-    } = await this._fetchIdentity();
-    const hostname = await this._fetchHost();
+  detect(_config?: ResourceDetectionConfig): Resource {
+    return new Resource({}, this._getAttributes());
+  }
 
-    return new Resource({
-      [SEMRESATTRS_CLOUD_PROVIDER]: CLOUDPROVIDERVALUES_ALIBABA_CLOUD,
-      [SEMRESATTRS_CLOUD_PLATFORM]: CLOUDPLATFORMVALUES_ALIBABA_CLOUD_ECS,
-      [SEMRESATTRS_CLOUD_ACCOUNT_ID]: accountId,
-      [SEMRESATTRS_CLOUD_REGION]: region,
-      [SEMRESATTRS_CLOUD_AVAILABILITY_ZONE]: availabilityZone,
-      [SEMRESATTRS_HOST_ID]: instanceId,
-      [SEMRESATTRS_HOST_TYPE]: instanceType,
-      [SEMRESATTRS_HOST_NAME]: hostname,
-    });
+  /** Gets identity and host info and returns them as attribs. Empty object if fails */
+  async _getAttributes(
+    _config?: ResourceDetectionConfig
+  ): Promise<ResourceAttributes> {
+    let attribs: ResourceAttributes;
+    try {
+      const {
+        'owner-account-id': accountId,
+        'instance-id': instanceId,
+        'instance-type': instanceType,
+        'region-id': region,
+        'zone-id': availabilityZone,
+      } = await this._fetchIdentity();
+      const hostname = await this._fetchHost();
+
+      attribs = {
+        [SEMRESATTRS_CLOUD_PROVIDER]: CLOUDPROVIDERVALUES_ALIBABA_CLOUD,
+        [SEMRESATTRS_CLOUD_PLATFORM]: CLOUDPLATFORMVALUES_ALIBABA_CLOUD_ECS,
+        [SEMRESATTRS_CLOUD_ACCOUNT_ID]: accountId,
+        [SEMRESATTRS_CLOUD_REGION]: region,
+        [SEMRESATTRS_CLOUD_AVAILABILITY_ZONE]: availabilityZone,
+        [SEMRESATTRS_HOST_ID]: instanceId,
+        [SEMRESATTRS_HOST_TYPE]: instanceType,
+        [SEMRESATTRS_HOST_NAME]: hostname,
+      };
+    } catch {
```

Comment:
Could a diag log be added for the error?

Reply:
Given that the rejection is handled at https://github.com/open-telemetry/opentelemetry-js/blob/54b14fbbe33e0cf38115ee8c5c934a4e27786186/packages/opentelemetry-resources/src/Resource.ts#L87-L91. I think we should propagate the error instead of discarding it here.

Reply:
I agree.